tests/contract.rs

```diff
@@ -33,18 +34,24 @@ fn contract_tests(file_path: &str, target: Target) -> io::Result<()> {
 }
 
 fn recurse_directory(path: PathBuf, target: Target) -> io::Result<()> {
+    let mut entries = Vec::new();
+
     for entry in read_dir(path)? {
         let path = entry?.path();
 
         if path.is_dir() {
             recurse_directory(path, target)?;
         } else if let Some(ext) = path.extension() {
             if ext.to_string_lossy() == "sol" {
-                parse_file(path, target)?;
+                entries.push(path);
```

Comment:
Why is this change necessary?

Reply:
I changed the test code so it uses `rayon` to use all available cores. This makes the test much faster

Reply:
We already run Solana contracts, Substrate contracts and EVM contracts tests in parallel as they are different `#[test]` functions functions. Having another layer of parallelism here may slow down the tests (I have not tested though).